### ISSUE TITLE
Crock and Auction house Client side crash

### ISSUE THREAD

**Issue by Darkx352** [anegostudios/VintageStory-Issues#4709]
### Game Version

v1.20.rc4

### Platform

Windows

### Modded

Modded

### SP/MP

Multiplayer

### Description

If you put a crock in the AH to sell with food in it. When the person goes to buy it. It will crash their client. 

### How to reproduce

First, put a crock in the auction house with food in it. 
a second person buys the crock pot with food in it. 
as soon as you hit buy you crash

### Logs
```
Running on 64 bit Windows 10.0.22631.0 with 130987 MB RAM
Game Version: v1.20.0-rc.4 (Unstable)
12/15/2024 7:49:15 PM: Critical error occurred
Loaded Mods: ClaimsRadar@1.1.0, moreicons@1.1.0, visibleore@1.0.1, voicevolume@1.1.0, game@1.20.0-rc.4, vsimgui@1.1.7, zoombuttonreborn@2.0.0, alloycalculator@1.0.0, billygoat@0.0.1, helblockpick@2.4.0, chattranslator@1.1.0, configlib@1.3.17, creaturekilledby@1.0.0, danatweaks@3.3.5, daytrader@1.0.2, extrainfo@1.8.2, freedomunits@1.1.4, glowingarrows@1.2.2, hudclockpatch@1.0.0, knapster@2.11.2, particlesplus@1.1.0, playerlist@2.1.4, pomnotes@0.7.3-rc.1, targetDotMedium@1.1.0, translocatormarker@0.0.2, vmetp@1.1.1, vresp@1.1.1, creative@1.20.0-rc.4, survival@1.20.0-rc.4, statushudcont@2.1.2
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.GameContent.BlockCrock.<>c__DisplayClass18_0.<GetHeldItemInfo>b__0(EnumTransitionType transType, ItemStack stack, Single mul) in VSSurvivalMod\Block\BlockCrock.cs:line 420
   at Vintagestory.API.Common.CollectibleObject.GetTransitionRateMul(IWorldAccessor world, ItemSlot inSlot, EnumTransitionType transType) in VintagestoryApi\Common\Collectible\Collectible.cs:line 2429
   at Vintagestory.API.Common.CollectibleObject.UpdateAndGetTransitionStatesNative(IWorldAccessor world, ItemSlot inslot) in VintagestoryApi\Common\Collectible\Collectible.cs:line 2457
   at Vintagestory.API.Common.CollectibleObject.AppendPerishableInfoText(ItemSlot inSlot, StringBuilder dsc, IWorldAccessor world) in VintagestoryApi\Common\Collectible\Collectible.cs:line 1804
   at Vintagestory.GameContent.BlockCrock.GetHeldItemInfo(ItemSlot inSlot, StringBuilder dsc, IWorldAccessor world, Boolean withDebugInfo) in VSSurvivalMod\Block\BlockCrock.cs:line 472
   at Vintagestory.API.Common.ItemStack.GetDescription(IWorldAccessor world, ItemSlot inSlot, Boolean debug) in VintagestoryApi\Common\Collectible\ItemStack.cs:line 399
   at Vintagestory.GameContent.GuiDialogConfirmPurchase.Init() in VSSurvivalMod\Systems\Auction\GuiDialogConfirmPurchase.cs:line 59
   at Vintagestory.GameContent.GuiDialogConfirmPurchase..ctor(ICoreClientAPI capi, EntityAgent buyerEntity, EntityAgent auctioneerEntity, Auction auction) in VSSurvivalMod\Systems\Auction\GuiDialogConfirmPurchase.cs:line 30
   at Vintagestory.GameContent.GuiDialogTrader.OnBuyAuctionClicked() in VSSurvivalMod\Gui\GuiDialogTrader.cs:line 276
   at Vintagestory.API.Client.GuiElementTextButton.OnMouseUpOnElement(ICoreClientAPI api, MouseEvent args) in VintagestoryApi\Client\UI\Elements\Impl\Interactive\Controls\GuiElementTextButton.cs:line 318
   at Vintagestory.API.Client.GuiElementTextButton.OnMouseUp(ICoreClientAPI api, MouseEvent args) in VintagestoryApi\Client\UI\Elements\Impl\Interactive\Controls\GuiElementTextButton.cs:line 308
   at Vintagestory.API.Client.GuiComposer.OnMouseUp(MouseEvent mouse) in VintagestoryApi\Client\UI\GuiComposer.cs:line 451
   at Vintagestory.API.Client.GuiDialog.OnMouseUp(MouseEvent args) in VintagestoryApi\Client\UI\Dialog\GuiDialog.cs:line 588
   at Vintagestory.Client.NoObf.GuiManager.OnMouseUp(MouseEvent args) in VintagestoryLib\Client\Systems\Gui\GuiManager.cs:line 412
   at Vintagestory.Client.NoObf.ClientMain.UpdateMouseButtonState(EnumMouseButton button, Boolean down) in VintagestoryLib\Client\ClientMain.cs:line 1924
   at Vintagestory.Client.HotkeyManager.TriggerHotKey(KeyEvent keyEventargs, IWorldAccessor world, IPlayer player, Boolean allowCharacterControls, Boolean isGlobal, Boolean fallBack, Boolean keyup) in VintagestoryLib\Client\HotkeyManager.cs:line 431
   at Vintagestory.Client.GuiScreenRunningGame.OnMouseUp(MouseEvent args) in VintagestoryLib\Client\MainMenu\Screens\GuiScreenRunningGame.cs:line 352
   at Vintagestory.Client.NoObf.ClientPlatformWindows.Mouse_ButtonUp(MouseButtonEventArgs e) in VintagestoryLib\Client\ClientPlatform\Input.cs:line 208
   at OpenTK.Windowing.Desktop.NativeWindow.MouseButtonCallback(Window* window, MouseButton button, InputAction action, KeyModifiers mods)
--- End of stack trace from previous location ---
   at OpenTK.Windowing.Desktop.NativeWindow.RethrowCallbackExceptionsIfNeeded()
   at OpenTK.Windowing.Desktop.GameWindow.Run()
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs) in VintagestoryLib\Client\ClientProgram.cs:line 334
   at Vintagestory.Client.ClientProgram.<>c__DisplayClass10_0.<.ctor>b__1() in VintagestoryLib\Client\ClientProgram.cs:line 130
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93

Event Log entries for Vintagestory.exe, the latest 3
==================================
{ TimeGenerated = 12/15/2024 2:35:49 PM, Site = , Source = Application Error, Message = Faulting application name: Vintagestory.exe, version: 1.20.0.0, time stamp: 0x66470000
Faulting module name: openal32.dll, version: 1.23.0.0, time stamp: 0x63dd31ad
Exception code: 0x40000015
Fault offset: 0x00000000000df046
Faulting process id: 0x0xabfc
Faulting application start time: 0x0x1db4f2844754fa0
Faulting application path: C:\Users\New\AppData\Roaming\VS1.20.rc4\Vintagestory.exe
Faulting module path: C:\Users\New\AppData\Roaming\VS1.20.rc4\Lib\openal32.dll
Report Id: f77c2cff-6341-44c1-8f2a-efe420cb828d
Faulting package full name: 
Faulting package-relative application ID:  }
--------------
{ TimeGenerated = 12/15/2024 2:33:49 PM, Site = , Source = Application Error, Message = Faulting application name: Vintagestory.exe, version: 1.20.0.0, time stamp: 0x66470000
Faulting module name: openal32.dll, version: 1.23.0.0, time stamp: 0x63dd31ad
Exception code: 0x40000015
Fault offset: 0x00000000000df046
Faulting process id: 0x0xbbe8
Faulting application start time: 0x0x1db4f27d9e5e42a
Faulting application path: C:\Users\New\AppData\Roaming\VS1.20.rc4\Vintagestory.exe
Faulting module path: C:\Users\New\AppData\Roaming\VS1.20.rc4\Lib\openal32.dll
Report Id: 683a7c58-27e1-4cfb-9854-f5e5ce44f52c
Faulting package full name: 
Faulting package-relative application ID:  }
--------------
{ TimeGenerated = 12/12/2024 7:48:34 PM, Site = , Source = Application Error, Message = Faulting application name: Vintagestory.exe, version: 1.20.0.0, time stamp: 0x66470000
Faulting module name: openal32.dll, version: 1.23.0.0, time stamp: 0x63dd31ad
Exception code: 0x40000015
Fault offset: 0x00000000000df046
Faulting process id: 0x0x8f18
Faulting application start time: 0x0x1db4cf886c9c1d4
Faulting application path: C:\Users\New\AppData\Roaming\VintagestoryData\VSTestServer\Vintagestory.exe
Faulting module path: C:\Users\New\AppData\Roaming\VintagestoryData\VSTestServer\Lib\openal32.dll
Report Id: 99defd75-ba8a-4cfe-a61c-9a51b180fbc1
Faulting package full name: 
Faulting package-relative application ID:  }
```

**Comment by Craluminum2413** [comment]
Caused by mod, no idea which one. Try without mods

**Comment by Darkx352** [comment]
No mods installed. 

Running on 64 bit Windows 10.0.22631.0 with 130987 MB RAM
Game Version: v1.20.0-rc.4 (Unstable)
12/15/2024 7:49:15 PM: Critical error occurred
Loaded Mods: ClaimsRadar@1.1.0, moreicons@1.1.0, visibleore@1.0.1, voicevolume@1.1.0, game@1.20.0-rc.4, vsimgui@1.1.7, zoombuttonreborn@2.0.0, alloycalculator@1.0.0, billygoat@0.0.1, helblockpick@2.4.0, chattranslator@1.1.0, configlib@1.3.17, creaturekilledby@1.0.0, danatweaks@3.3.5, daytrader@1.0.2, extrainfo@1.8.2, freedomunits@1.1.4, glowingarrows@1.2.2, hudclockpatch@1.0.0, knapster@2.11.2, particlesplus@1.1.0, playerlist@2.1.4, pomnotes@0.7.3-rc.1, targetDotMedium@1.1.0, translocatormarker@0.0.2, vmetp@1.1.1, vresp@1.1.1, creative@1.20.0-rc.4, survival@1.20.0-rc.4, statushudcont@2.1.2
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.GameContent.BlockCrock.<>c__DisplayClass18_0.<GetHeldItemInfo>b__0(EnumTransitionType transType, ItemStack stack, Single mul) in VSSurvivalMod\Block\BlockCrock.cs:line 420
   at Vintagestory.API.Common.CollectibleObject.GetTransitionRateMul(IWorldAccessor world, ItemSlot inSlot, EnumTransitionType transType) in VintagestoryApi\Common\Collectible\Collectible.cs:line 2429
   at Vintagestory.API.Common.CollectibleObject.UpdateAndGetTransitionStatesNative(IWorldAccessor world, ItemSlot inslot) in VintagestoryApi\Common\Collectible\Collectible.cs:line 2457
   at Vintagestory.API.Common.CollectibleObject.AppendPerishableInfoText(ItemSlot inSlot, StringBuilder dsc, IWorldAccessor world) in VintagestoryApi\Common\Collectible\Collectible.cs:line 1804
   at Vintagestory.GameContent.BlockCrock.GetHeldItemInfo(ItemSlot inSlot, StringBuilder dsc, IWorldAccessor world, Boolean withDebugInfo) in VSSurvivalMod\Block\BlockCrock.cs:line 472
   at Vintagestory.API.Common.ItemStack.GetDescription(IWorldAccessor world, ItemSlot inSlot, Boolean debug) in VintagestoryApi\Common\Collectible\ItemStack.cs:line 399
   at Vintagestory.GameContent.GuiDialogConfirmPurchase.Init() in VSSurvivalMod\Systems\Auction\GuiDialogConfirmPurchase.cs:line 59
   at Vintagestory.GameContent.GuiDialogConfirmPurchase..ctor(ICoreClientAPI capi, EntityAgent buyerEntity, EntityAgent auctioneerEntity, Auction auction) in VSSurvivalMod\Systems\Auction\GuiDialogConfirmPurchase.cs:line 30
   at Vintagestory.GameContent.GuiDialogTrader.OnBuyAuctionClicked() in VSSurvivalMod\Gui\GuiDialogTrader.cs:line 276
   at Vintagestory.API.Client.GuiElementTextButton.OnMouseUpOnElement(ICoreClientAPI api, MouseEvent args) in VintagestoryApi\Client\UI\Elements\Impl\Interactive\Controls\GuiElementTextButton.cs:line 318
   at Vintagestory.API.Client.GuiElementTextButton.OnMouseUp(ICoreClientAPI api, MouseEvent args) in VintagestoryApi\Client\UI\Elements\Impl\Interactive\Controls\GuiElementTextButton.cs:line 308
   at Vintagestory.API.Client.GuiComposer.OnMouseUp(MouseEvent mouse) in VintagestoryApi\Client\UI\GuiComposer.cs:line 451
   at Vintagestory.API.Client.GuiDialog.OnMouseUp(MouseEvent args) in VintagestoryApi\Client\UI\Dialog\GuiDialog.cs:line 588
   at Vintagestory.Client.NoObf.GuiManager.OnMouseUp(MouseEvent args) in VintagestoryLib\Client\Systems\Gui\GuiManager.cs:line 412
   at Vintagestory.Client.NoObf.ClientMain.UpdateMouseButtonState(EnumMouseButton button, Boolean down) in VintagestoryLib\Client\ClientMain.cs:line 1924
   at Vintagestory.Client.HotkeyManager.TriggerHotKey(KeyEvent keyEventargs, IWorldAccessor world, IPlayer player, Boolean allowCharacterControls, Boolean isGlobal, Boolean fallBack, Boolean keyup) in VintagestoryLib\Client\HotkeyManager.cs:line 431
   at Vintagestory.Client.GuiScreenRunningGame.OnMouseUp(MouseEvent args) in VintagestoryLib\Client\MainMenu\Screens\GuiScreenRunningGame.cs:line 352
   at Vintagestory.Client.NoObf.ClientPlatformWindows.Mouse_ButtonUp(MouseButtonEventArgs e) in VintagestoryLib\Client\ClientPlatform\Input.cs:line 208
   at OpenTK.Windowing.Desktop.NativeWindow.MouseButtonCallback(Window* window, MouseButton button, InputAction action, KeyModifiers mods)
--- End of stack trace from previous location ---
   at OpenTK.Windowing.Desktop.NativeWindow.RethrowCallbackExceptionsIfNeeded()
   at OpenTK.Windowing.Desktop.GameWindow.Run()
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs) in VintagestoryLib\Client\ClientProgram.cs:line 334
   at Vintagestory.Client.ClientProgram.<>c__DisplayClass10_0.<.ctor>b__1() in VintagestoryLib\Client\ClientProgram.cs:line 130
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93

Event Log entries for Vintagestory.exe, the latest 3
==================================
{ TimeGenerated = 12/15/2024 2:35:49 PM, Site = , Source = Application Error, Message = Faulting application name: Vintagestory.exe, version: 1.20.0.0, time stamp: 0x66470000
Faulting module name: openal32.dll, version: 1.23.0.0, time stamp: 0x63dd31ad
Exception code: 0x40000015
Fault offset: 0x00000000000df046
Faulting process id: 0x0xabfc
Faulting application start time: 0x0x1db4f2844754fa0
Faulting application path: C:\Users\New\AppData\Roaming\VS1.20.rc4\Vintagestory.exe
Faulting module path: C:\Users\New\AppData\Roaming\VS1.20.rc4\Lib\openal32.dll
Report Id: f77c2cff-6341-44c1-8f2a-efe420cb828d
Faulting package full name: 
Faulting package-relative application ID:  }
--------------
{ TimeGenerated = 12/15/2024 2:33:49 PM, Site = , Source = Application Error, Message = Faulting application name: Vintagestory.exe, version: 1.20.0.0, time stamp: 0x66470000
Faulting module name: openal32.dll, version: 1.23.0.0, time stamp: 0x63dd31ad
Exception code: 0x40000015
Fault offset: 0x00000000000df046
Faulting process id: 0x0xbbe8
Faulting application start time: 0x0x1db4f27d9e5e42a
Faulting application path: C:\Users\New\AppData\Roaming\VS1.20.rc4\Vintagestory.exe
Faulting module path: C:\Users\New\AppData\Roaming\VS1.20.rc4\Lib\openal32.dll
Report Id: 683a7c58-27e1-4cfb-9854-f5e5ce44f52c
Faulting package full name: 
Faulting package-relative application ID:  }
--------------
{ TimeGenerated = 12/12/2024 7:48:34 PM, Site = , Source = Application Error, Message = Faulting application name: Vintagestory.exe, version: 1.20.0.0, time stamp: 0x66470000
Faulting module name: openal32.dll, version: 1.23.0.0, time stamp: 0x63dd31ad
Exception code: 0x40000015
Fault offset: 0x00000000000df046
Faulting process id: 0x0x8f18
Faulting application start time: 0x0x1db4cf886c9c1d4
Faulting application path: C:\Users\New\AppData\Roaming\VintagestoryData\VSTestServer\Vintagestory.exe
Faulting module path: C:\Users\New\AppData\Roaming\VintagestoryData\VSTestServer\Lib\openal32.dll
Report Id: 99defd75-ba8a-4cfe-a61c-9a51b180fbc1
Faulting package full name: 
Faulting package-relative application ID:  }

**Comment by radfast** [comment]
It's a vanilla issue.  Fixed in next upcoming release (1.20.0-rc.5) when it comes.  Until then, it will unfortunately continue crashing if such items are bought in auctions, nothing can be done except not buy the items.
